crates/oxc_linter/src/rules/eslint/no_invalid_regexp.rs

```diff
@@ -0,0 +1,240 @@
+    fixer::{RuleFix, RuleFixer},
+    rule::Rule,
+    AstNode,
+};
+
+#[derive(Debug, Default, Clone)]
+pub struct NoInvalidRegexp;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    ///
+    /// ### Why is this bad?
+    ///
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```js
+    /// FIXME: Tests will fail if examples are missing or syntactically incorrect.
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// FIXME: Tests will fail if examples are missing or syntactically incorrect.
+    /// ```
+    NoInvalidRegexp,
+    nursery, // TODO: change category to `correctness`, `suspicious`, `pedantic`, `perf`, `restriction`, or `style`
+             // See <https://oxc.rs/docs/contribute/linter.html#rule-category> for details
+
+    pending  // TODO: describe fix capabilities. Remove if no fix can be done,
+             // keep at 'pending' if you think one could be added but don't know how.
+             // Options are 'fix', 'fix_dangerous', 'suggestion', and 'conditional_fix_suggestion'
+);
+
+impl Rule for NoInvalidRegexp {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let (pattern, flags, span_offset) = match node.kind() {
+            AstKind::NewExpression(e) => {
+                if !e.callee.is_specific_id("RegExp") {
+                    return;
+                }
+                let Some(Argument::StringLiteral(pattern)) = e.arguments.get(0) else { return };
+                let Some(Argument::StringLiteral(flags)) = e.arguments.get(1) else { return };
+                (pattern.value.as_str(), flags.value.as_str(), pattern.span.start)
+            }
+            AstKind::CallExpression(e) => {
+                if !e.callee.is_specific_id("RegExp") {
+                    return;
+                }
+                let Some(Argument::StringLiteral(pattern)) = e.arguments.get(0) else { return };
+                let Some(Argument::StringLiteral(flags)) = e.arguments.get(1) else { return };
+                (pattern.value.as_str(), flags.value.as_str(), pattern.span.start)
+            }
+            _ => return,
+        };
+
+        let allocator = Allocator::default();
+        let flags = match FlagsParser::new(&allocator, flags, ParserOptions::default()).parse() {
+            Ok(flags) => flags,
+            Err(diagnostic) => {
+                ctx.diagnostic(diagnostic);
+                return;
+            }
+        };
+
+        let options = ParserOptions {
+            span_offset: span_offset + 1,
+            unicode_mode: flags.unicode || flags.unicode_sets,
+            unicode_sets_mode: flags.unicode_sets,
+        };
+
+        let Err(diagnostic) = PatternParser::new(&allocator, pattern, options).parse() else {
+            return;
+        };
+
+        ctx.diagnostic(diagnostic);
+    }
+}
+
+#[test]
+fn test() {
+    use crate::tester::Tester;
+
+    let pass = vec![
```

Comment:
Some of these may have incorrect `\`, please double check with https://github.com/eslint/eslint/blob/main/tests/lib/rules/no-invalid-regexp.js